react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+const topics = [
+  { name: 'HTML', description: 'HTML (HyperText Markup Language) is the code that is used to structure a web page and its content. For example, content could be structured within a set of paragraphs, a list of bulleted points, or using images and data tables.' },
+  { name: 'CSS', description: 'Cascading Style Sheets is a style sheet language used for describing the presentation of a document written in a markup language such as HTML. ' },
+  { name: 'JavaScript', description: 'JavaScript, often abbreviated JS, is a programming language that is one of the core technologies of the World Wide Web, alongside HTML and CSS.' }
+];
+
+export default class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { selected: null };
+  }
+
+  handleClick(event) {
+    if (event.target.id === this.state.selected) {
+      this.setState({ selected: null });
+    } else (this.setState({ selected: event.target.id }));
+  }
+
+  render() {
+    return (
+      <ul className='container'>
+        {topics.map((topic, index) => {
+          if (index.toString() === this.state.selected) {
+            this.classVar = '';
+          } else { this.classVar = 'hidden'; }
```

Comment:
It's better to declare a variable for this class instead of assigning it to the `this` object.

```js
          let classVar;
          if (index.toString() === this.state.selected) {
           classVar = '';
          } else { classVar = 'hidden'; }
```